src/driver.ml

```diff
@@ -919,16 +861,28 @@ let process_file (kind : Kind.t) fn ~input_name ~relocate ~output_mode ~embed_er
              ~repl:(Many generated))
     }
   in
-  let input_name, ast =
-    try
-      let input_name, ast =
-        with_preprocessed_input fn ~f:(load_input kind fn input_name ~relocate)
-      in
-      let ast = extract_cookies ast in
-      process_ast ast ~input_name ~tool_name ~hook ~expect_mismatch_handler
-    with exn when embed_errors -> handle_exn exn ~input_name ~kind
-  in
 
+  let input_name, input_version, ast =
+    match
+      with_preprocessed_input fn ~f:(load_input kind fn input_name ~relocate)
+    with
+    | Ok (input_fname, input_version, ast) -> (
+        try
+          let ast =
+            extract_cookies ast
+            |> process_ast ~tool_name ~hook ~expect_mismatch_handler
+          in
+          (input_fname, input_version, ast)
+        with exn when embed_errors ->
+          (input_fname, input_version, exn_to_extension exn ~kind) )
+    | Error (error, input_version) when embed_errors ->
+        (input_name, input_version, error_to_extension error ~kind)
+    | Error (error, _) ->
+        Location.raise_errorf
+          ~loc:(Location.in_file input_name)
+          "%s"
+          (Location.Error.message error)
```

Comment:
Is there a better way to do this?

Reply:
What about adding a `Location.Error.raise` function?

Reply:
Yes, good idea! I've done that now. The only downside that has is that if the initial exception was raised without `loc` parameter, here I can't add it anymore. But that only has an impact for `input doesn't exist` errors (see [here](https://github.com/ocaml-ppx/ppxlib/pull/205/files#diff-42ed2efe32ada9a3626b9feb98bca6c1e2643cb2e07c77d8c3f965a3022458e2R54)), where the input file is reported inside the error message anyways.